modules/internal/ChapelArray.chpl

```diff
@@ -1885,6 +1885,50 @@ module ChapelArray {
     pragma "no doc"
     proc indexOrder(i) return _value.dsiIndexOrder(_makeIndexTuple(rank, i));
 
+    /*
+       Returns domain index for a given order in the domain
```

Comment:
Let's reword this line and add some simple examples. Here's a suggested text:

```
Returns the `ith` index in the domain. For example, the `2nd` index of ``{2..10 by 2}`` would be ``4``.

The order of multidimensional domains follows their 1-dimensional iterator.
For example, the `4th` index of ``{1..3, 1..2}`` would be ``(2,1)``.
```

Reply:
"1-dimensional iterator for multidimensional domains" does not ring a bell to me.
Did you mean "serial iterator" ?

Minor: let us go with singular: `The order of [for?] a multidimensional domain follows` ......

Reply:
>Did you mean "serial iterator"

Yes, thanks for catching that.

@CaptainSharf - combining @vasslitvinov's feedback into my previous suggestion:

```
Returns the `ith` index in the domain. For example, the `2nd` index of ``{2..10 by 2}`` would be ``4``.

The order of a multidimensional domain follows the serial iterator.
For example, the `4th` index of ``{1..3, 1..2}`` would be ``(2,1)``.
```

Reply:
@ben-albrecht , wouldn't the `orderToIndex` for the second domain be `(2,2)` , as we're using 0 based indexing here?

Reply:
I also wanted to verify the behavior of this method when the rank of the domain is 1. In the first example, the method returns a 1-tuple i.e `(6)` .  Would this be acceptable?

Reply:
> @ben-albrecht , wouldn't the `orderToIndex` for the second domain be `(2,2)` , as we're using 0 based indexing here?

Oops, yes you are right. Good catch! I suppose the test would catch this too :)

Reply:
> In the first example, the method returns a 1-tuple i.e (6) . Would this be acceptable?

Good question -- I was intuitively expecting an integer when rank is 1. Is there a rationale for making it a 1-tuple? Maybe we ought to bring this up in https://github.com/chapel-lang/chapel/issues/14885.

Reply:
Sounds like there is consensus to always returning the `idxType` in https://github.com/chapel-lang/chapel/issues/14885. This means rank 1 domains just return the `idxType` instead of a 1-tuple, e.g. `6`, not `(6,)`.